src/pages/MainPage/MainPage.tsx

```diff
@@ -0,0 +1,107 @@
+import './MainPage.css';
+import { useEffect, useMemo, useState } from 'react';
+import { Pet } from '../../types/pet.ts';
+import { getPets } from '../../api-repositories/pets/pets.ts';
+import { formatSearchInput } from '../../utils/formatSearchInput.ts';
+import { PetCard } from '../../components/PetCard/PetCard.tsx';
+import { TopControls } from './TopControls/TopControls.tsx';
+import { Spinner } from '../../components/Spinner/Spinner.tsx';
+import { Pagination } from '../../components/Pagination/Pagination.tsx';
+import { Link, Outlet, useLocation, useNavigation, useSearchParams } from 'react-router-dom';
+import { useSearchTerm } from './useSearchTerm.hook.ts';
+import { getDetailedPetPagePath } from '../../router/router.tsx';
+
+const ITEMS_PER_PAGE = 10;
+export const SEARCH_PARAMS_PAGE = 'page';
+
+export function MainPage() {
+  const [pets, setPets] = useState<Pet[] | undefined>(undefined);
+  const [isLoading, setIsLoading] = useState<boolean>(false);
+  const [searchParams, setSearchParams] = useSearchParams();
+  const [totalPages, setTotalPages] = useState<number | undefined>(undefined);
+  const { searchTerm, setSearchTerm } = useSearchTerm();
+  const location = useLocation();
+  const activePageSearchParam = searchParams.get(SEARCH_PARAMS_PAGE);
+  const navigation = useNavigation();
+
+  const activePage = useMemo(() => {
+    const activePage = parseInt(activePageSearchParam || '', 10);
+    return Number.isInteger(activePage) ? activePage : 1;
+  }, [activePageSearchParam]);
+
+  const setActivePage = (val: number) => {
+    const newUrlSearchParams = new URLSearchParams(searchParams);
+    newUrlSearchParams.set(SEARCH_PARAMS_PAGE, `${val}`);
+    setSearchParams(newUrlSearchParams);
+  };
+
+  const fetchPets = async (params: { name: string; offset?: number }) => {
+    const { name, offset } = params;
+
+    try {
+      setIsLoading(true);
+
+      const { totalPages, pets } = await getPets({
+        name: formatSearchInput(name),
+        pageNumber: offset,
+        pageSize: ITEMS_PER_PAGE,
+      });
+
+      setPets(pets);
+      setTotalPages(totalPages);
+      setIsLoading(false);
+    } catch {
+      setIsLoading(false);
+    }
```

Comment:
Consider using finally section here 
```suggestion
    } catch {
      // handle an error
    } finally {
      setIsLoading(false);
    }
```